mission_control/src/behaviors/attitude_servo.cpp

```diff
@@ -88,14 +90,19 @@ BT::NodeStatus AttitudeServoBehavior::behaviorRunningProcess()
   if (!goalHasBeenPublished_)
   {
     publishGoalMsg();
+    behaviorStartTime_ = ros::Time::now();
     goalHasBeenPublished_ = true;
   }
   else
   {
     ros::Duration delta_t = ros::Time::now() - behaviorStartTime_;
     if (delta_t.toSec() > timeOut_) setStatus(BT::NodeStatus::FAILURE);
+    else
+    {
+      if (behaviorComplete_) setStatus(BT::NodeStatus::SUCCESS);
```

Comment:
@patapufate consider checking for completion before checking for timeout.

Reply:
Also, meta: stateful actions like these won't allow mission re-execution.

Reply:
Yes. Good catch. That's why I put behaviorStartTime_ when we publish the goal.